### DISCUSSION TITLE
previews/pr-20/blogs/2026-004/

### DISCUSSION THREAD

**Discussion by giscus** [genomicsxai/genomicsxai.github.io#21] [Post Discussions]
# previews/pr-20/blogs/2026-004/

 Summary This post introduces alphagenome-pytorch, a faithful reimplementation of Google DeepMind’s AlphaGenome model in PyTorch.\nWe reproduce the full AlphaGenome architecture in PyTorch and release PyTorch weights for fold-specific and distilled models. We also verify numerical equivalence of predictions across all output tracks with the JAX checkpoint from DeepMind. Our package exposes a simple inference API that slots into any PyTorch project without requiring JAX, XLA, or TPU-specific tooling.\nIn this post, we review the initial release for the package and walk through two use cases it enables:\n

https://genomicsxai.github.io/previews/pr-20/blogs/2026-004/?v=2

**Comment by balwierz** [comment]
Great work!
Do you know what are the hardware requirements (GPU VRAM) to run it?

**Reply by ejarmand** [reply]
Depends on sequence length. I've been playing with a 524kb input sequence (matching borzoi), and I can fit a batch size of 8 in ~60 gb vram (80gb A100, bfloat16, ~1600 output tracks). 

Can probably forward on something pretty small if you're doing bfloat 16 (e.g. 12-16 gb), and I'd expect on 48gb vram you can train at full length will a small batch size (and maybe some tricks with gradient checkpointing if you use the unet).

**Reply by ejarmand** [reply]
To clarify, I trained with 128bp outputs/embeddings.

**Reply by Al-Murphy** [reply]
@balwierz also see this blog post for a detailed breakdown of hardware requirements: https://genomicsxai.github.io/blogs/2026-005/

**Comment by ArghamitraT** [comment]
I wanted to let you know how amazingly easy this work has made my life. My HPC uses an AMD system so I would have to put a bit of work to make the JAX version work. Not only the pytorch, the git repo, documentations, notebook examples make it convenient to use. 

Thank you!
